docs/table/construct_table.rst

```diff
@@ -960,3 +964,71 @@ a `~astropy.table.Column` object with a ``unit`` attribute::
 
 To learn more about using standard `~astropy.table.Column` objects with defined
 units, see the :ref:`columns_with_units` section.
+
+
+Compatible tables
+^^^^^^^^^^^^^^^^^
+
+In order to improve interoperability between different table classes, an
+astropy |Table| object can be created directly from any other table
+object ``data`` that provides an ``__astropy_table__`` method.  In this case the
+``__astropy_table__`` method will be called as follows::
+
+  table = data.__astropy_table__(cls, copy)
+
+where ``cls`` is the |Table| class or subclass that is being instantiated,
+and ``copy`` indicates whether a copy of the values in ``data`` should be
+provided.  If ``copy`` is ``False`` then the ``__astropy_table__`` method
```

Comment:
Interesting.  Maybe astropy.Table has been too conservative in the interpretation.  It definitely always copies when `copy=True` but in at least some circumstances it purposely raises an exception when `copy=False` but it can't return a reference.  Maybe we should make this more liberal into the "return a reference if you can"?  In fact `Table` is somewhat inconsistent about applying the rule.

So I will change the definition accordingly, and probably open an issue to make `Table` use that interpretation.